src/components/Button/Button.tsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import './Button.css';
```

Comment:
```suggestion
import сss from './Button.css';
```

и затем использовать в компоненте

```Typescript
<button type="button" {...buttonProps} className={css.button}>
```

При таком подходе провалидируется, что не было ошибки в имени класса. 

Правда, при написании имен стилей придется перейти на `camelCase`, чтобы не писать в коде `css['input__field']`. Либо поискать плагин для вебпака или `postCss`,  который перепишет имена в `camelCase` стиле